kyaml/yaml/rnode.go

```diff
@@ -546,6 +546,29 @@ func (rn *RNode) GetBinaryDataMap() map[string]string {
 	return result
 }
 
+// GetValidatedDataMap retrieves the data map and returns an error if the data
+// map contains entries which are not included in the expected set
+func (rn *RNode) GetValidatedDataMap(expected []string) (map[string]string, error) {
+	dataMap := rn.GetDataMap()
+	err := rn.validateDataMap(dataMap, expected)
+	return dataMap, err
+}
+
+func (rn *RNode) validateDataMap(dataMap map[string]string, expectedKeys []string) error {
```

Comment:
You need to check if `dataMap` is nil, since `rn.GetDataMap()` may return nil. IMO when `dataMap` is nil, we should fail.
This will happen when the object doesn't have a `data` field.

Reply:
`GetDataMap` returns an empty map if `data` is not present. That case is tested here:

https://github.com/kubernetes-sigs/kustomize/blob/23e28bb18ad9dbcc543c144b3266d04119063ec1/kyaml/yaml/rnode_test.go#L141-L147

Reply:
Got it.